arangod/Cluster/ResignShardLeadership.cpp

```diff
@@ -130,15 +119,31 @@ bool ResignShardLeadership::first() {
     // for now but we will not accept any replication operation from any
     // leader, until we have negotiated a deal with it. Then the actual
     // name of the leader will be set.
-    col->followers()->setTheLeader(LeaderNotYetKnownString);  // resign
-    res = methods->abort();                                   // unlock
+    for (auto const& s : shards) {
+      auto col = vocbase->lookupCollection(s);
+      if (col == nullptr) {
+        std::stringstream error;
+        error << "Failed to lookup local collection " << collection
+              << " in database " << database;
+        LOG_TOPIC("e06ca", ERR, Logger::MAINTENANCE)
+            << "ResignLeadership: " << error.str();
+        result(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, error.str());
+        return false;
+      }
+
+      col->followers()->setTheLeader(LeaderNotYetKnownString);  // resign
```

Comment:
If we have multiple shards in `shards` and loop has multiple iterations, we will call `setTheLeader(LeaderNotYetKnownString)` for all shards until we run into an error and abort the loop with `false`.
So may have resign for some shards but not others. While this may be perfectly fine, I don't know if it actually is and if there is some way that this mixed state is automatically resolved later.